include/Probe.php

```diff
@@ -0,0 +1,1114 @@
+				$path = str_replace('{uri}', urlencode($uri), $link);
+				$webfinger = self::webfinger($path);
+
+				// Since the detection with the address wasn't successful, we delete it.
+				if ($webfinger) {
+					$nick = "";
+					$addr = "";
+				}
+			}
+
+		}
+		if (!$webfinger)
+			return self::feed($uri);
+
+		$result = false;
+
+		logger("Probing ".$uri, LOGGER_DEBUG);
+
+		if (in_array($network, array("", NETWORK_DFRN)))
+			$result = self::dfrn($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_DIASPORA))
+			$result = self::diaspora($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_OSTATUS))
+			$result = self::ostatus($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_PUMPIO))
+			$result = self::pumpio($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_FEED))
+			$result = self::feed($uri);
+		else {
+			// We overwrite the detected nick with our try if the previois routines hadn't detected it.
+			// Additionally it is overwritten when the nickname doesn't make sense (contains spaces).
+			if ((!isset($result["nick"]) OR ($result["nick"] == "") OR (strstr($result["nick"], " "))) AND ($nick != ""))
+				$result["nick"] = $nick;
+
+			if ((!isset($result["addr"]) OR ($result["addr"] == "")) AND ($addr != ""))
+				$result["addr"] = $addr;
+		}
+
+		logger($uri." is ".$result["network"], LOGGER_DEBUG);
+
+		if (!isset($result["baseurl"]) OR ($result["baseurl"] == "")) {
+			$pos = strpos($result["url"], $host);
+			if ($pos)
+				$result["baseurl"] = substr($result["url"], 0, $pos).$host;
+		}
+
+		return $result;
+	}
+
+	/**
+	 * @brief Do a webfinger request
+	 *
+	 * @param string $url Address that should be probed
+	 *
+	 * @return array webfinger data
+	 */
+	private function webfinger($url) {
+
+		$xrd_timeout = Config::get('system','xrd_timeout', 20);
+		$redirects = 0;
+
+		$data = fetch_url($url, false, $redirects, $xrd_timeout, "application/xrd+xml");
+		$xrd = parse_xml_string($data, false);
+
+		if (!is_object($xrd)) {
+			// If it is not XML, maybe it is JSON
+			$webfinger = json_decode($data, true);
+
+			if (!isset($webfinger["links"]))
+				return false;
+
+			return $webfinger;
+		}
+
+		$xrd_arr = xml::element_to_array($xrd);
+		if (!isset($xrd_arr["xrd"]["link"]))
+			return false;
+
+		$webfinger = array();
+
+		if (isset($xrd_arr["xrd"]["subject"]))
+			$webfinger["subject"] = $xrd_arr["xrd"]["subject"];
+
+		if (isset($xrd_arr["xrd"]["alias"]))
+			$webfinger["aliases"] = $xrd_arr["xrd"]["alias"];
+
+		$webfinger["links"] = array();
+
+		foreach ($xrd_arr["xrd"]["link"] AS $value => $data) {
+			if (isset($data["@attributes"]))
+				$attributes = $data["@attributes"];
+			elseif ($value == "@attributes")
+				$attributes = $data;
+			else
+				continue;
+
+			$webfinger["links"][] = $attributes;
+		}
+		return $webfinger;
+	}
+
+	/**
+	 * @brief Poll the noscrape page (Friendica specific)
```

Comment:
Would need a short description what "nosrape" is. Something like: Wit noscrape we can get friendica profile information via json request (for further information have a look at /mod/noscrape.php)